Description: Backport to 2.6.18
 The function random32 was first introduced in v2.6.19-rc2-109-gaaa248f were it
 was renamed from net_random.
 .
 The in-kernel socket api was first introduced in v2.6.18-310-gac5a488. It is
 leightweight enough to implement needed functions as wrapper.
Forwarded: no
Author: Sven Eckelmann <sven.eckelmann@gmx.de>

---
diff --git a/compat.h b/compat.h
index 0ea2655d564238307df3edbe93cabf7f4d78ad27..53c1749b5dc2d538cef1458acaaa922541aa3be9 100644
--- a/compat.h
+++ b/compat.h
@@ -22,6 +22,15 @@
 
 #include <linux/version.h>	/* LINUX_VERSION_CODE */
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 19)
+
+#define random32() net_random()
+
+#define kernel_bind(_sock, _addr, _addrlen) \
+	_sock->ops->bind(_sock, _addr, _addrlen)
+
+#endif /* LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 19) */
+
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 20)
 
 #define delayed_work work_struct
